src/core/string_set.cc

```diff
@@ -0,0 +1,103 @@
+#include <cstdint>
+
+#include "core/compact_object.h"
+#include "core/string_set.h"
+#include "redis/sds.h"
+
+extern "C" {
+#include "redis/zmalloc.h"
+}
+
+namespace dfly {
+
+uint64_t StringSet::Hash(const void* ptr) const {
+  sds s = (sds)ptr;
+  return CompactObj::HashCode(std::string_view{s, sdslen(s)});
+}
+
+bool StringSet::Equal(const void* ptr1, const void* ptr2) const {
+  sds s1 = (sds)ptr1;
+  sds s2 = (sds)ptr2;
+
+  if (sdslen(s1) != sdslen(s2)) {
+    return false;
+  }
+  return sdslen(s1) == 0 || memcmp(s1, s2, sdslen(s1)) == 0;
+}
+
+size_t StringSet::ObjectAllocSize(const void* s1) const {
+  return zmalloc_usable_size(sdsAllocPtr((sds)s1));
+}
+
+bool StringSet::AddRaw(sds s1) {
```

Comment:
you prefer `Raw` suffix over `Sds` ? why?

Reply:
No specific reason the thought process was mainly `sds` = `char*` which is the raw pointer of a `string_view`